JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -196,294 +199,295 @@ TTestIndependentSamples <- function(jaspResults, dataset = NULL, options, ...) {
-  
-  variables  <- options$variables
-  if (length(variables) == 0) 
-    variables <- "."
+  if (options$hypothesis == "groupOneGreater" || options$hypothesis == "groupTwoGreater") {
+    directionNote <- ifelse(options$hypothesis == "groupOneGreater", gettext("greater"), gettext("less"))
+    table$addFootnote(gettextf("For all tests, the alternative hypothesis specifies that group %1$s is %2$s than group %3$s.",
+                                                paste("<em>", levels[1], "</em>"), directionNote, paste("<em>", levels[2], "</em>")))
+  }
   
-  if(!ready) {
-    for (variable in variables) 
-      jaspResults[["ttest"]]$addRows(list(v = variable))
-  } else {
-    levels <- levels(dataset[[ .v(options$groupingVariable) ]])
+  ## for each variable specified, run each test that the user wants
+  for (variable in options$variables) {
+    errors <- .hasErrors(dataset, 
+                         message = 'short', 
+                         type = c('observations', 'variance', 'infinity'),
+                         all.target = variable, 
+                         all.grouping = options$groupingVariable,
+                         observations.amount = '< 2')
     
-    if (options$hypothesis == "groupOneGreater" || options$hypothesis == "groupTwoGreater") {
-      directionNote <- ifelse(options$hypothesis == "groupOneGreater", gettext("greater"), gettext("less"))
-      jaspResults[["ttest"]]$addFootnote(gettextf("For all tests, the alternative hypothesis specifies that group %1$s is %2$s than group %3$s.",
-                                                  paste("<em>", levels[1], "</em>"), directionNote, paste("<em>", levels[2], "</em>")))
-    }
-    
-    groupingData <- dataset[[ .v(options$groupingVariable) ]]
-    ## for each variable specified, run each test that the user wants
-    for (variable in options$variables) {
-      errors <- .hasErrors(dataset, 
-                           message = 'short', 
-                           type = c('observations', 'variance', 'infinity'),
-                           all.target = variable, 
-                           all.grouping = options$groupingVariable,
-                           observations.amount = '< 2')
-      variableData <- dataset[[ .v(variable) ]]
+    ## test is a number, indicating which tests should be run
+    for (test in seq_len(length(optionsList$whichTests))) {
       
-      ## test is a number, indicating which tests should be run
-      for (test in seq_len(length(optionsList$whichTests))) {
-        
-        ## don't run a test the user doesn't want
-        currentTest <- optionsList$whichTests[[test]]
-        if (!currentTest)
-          next
+      # TODO Tim: whichTests should just be a character vector of test names requested. Then you lose the confusion of figuring out which test is which int and you don't need this check
+      ## don't run a test the user doesn't want
+      currentTest <- optionsList$whichTests[[test]]
```

Comment:
(Unsolicited comment as you probably know best how to do this :D)
Perhaps a lot of this can be simplified if inside of `.ttestOptionsList()`, you would define something like:

```
    optionsList$whichTests  <- list("1" = optionsList$wantsStudents, 
                                    "2" = optionsList$wantsWelchs, 
                                    "3" = optionsList$wantsWilcox)

    optionsList$requestedTests <- c(“Student”, “Welch”, “Mann-Whitney”)[unlist(optionsList$whichTests)]
```

Then you can just loop over the requested tests when appropriate, enabling you to skip all of these `next` statements and defining `whatTest`.

Reply:
Yeah, totally. I just got lazy, because my changes incited more changes, which incited changes. I figured I needed to draw the line somewhere. But alas! You have convinced me :-) I will get rid of the TODO and actually do it.